### PR TITLE
Update CHANGELOG.md to reflect 5.9 scope

### PR DESCRIPTION
**Pull request by tomerd**
motivation: improve CHANGELOG.md

changes:
* add missing features introduced in 5.9
* reorder list chronologically

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,25 +3,30 @@ Note: This is in reverse chronological order, so newer entries are added to the
...
 Swift 5.9
 -----------
 
+* [SE-0386]
+
+  SwiftPM packages can now use `package` as a new access modifier, allowing accessing symbols in another target / module within the same package without making it public.
+
+* [SE-0391]
+
+  SwiftPM can now publish to a registry following the pulish spec defined in [SE-0391]
```
Comment (tomerd) [inline CHANGELOG.md:20]:
```suggestion
  SwiftPM can now publish to a registry following the publishing spec as defined in [SE-0391]
```

**Comment by neonichu** [comment]
Should we add something for https://developer.apple.com/documentation/packagedescription/swiftsetting/interoperabilitymode(_:_:)/?

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,25 +3,30 @@ Note: This is in reverse chronological order, so newer entries are added to the
...
 Swift 5.9
 -----------
 
+* [SE-0386]
+
+  SwiftPM packages can now use `package` as a new access modifier, allowing accessing symbols in another target / module within the same package without making it public.
+
+* [SE-0391]
+
+  SwiftPM can now publish to a registry following the publishing spec as defined in [SE-0391]
```
Comment (yim-lee) [inline CHANGELOG.md:20]:
```suggestion
  SwiftPM can now publish to a registry following the publishing spec as defined in [SE-0391]. SwiftPM also gains support for signed packages. Trust-on-first-use (TOFU) check which includes only fingerprints (e.g., checksums) previously has been extended to include signing identities, and it is enforced for source archives as well as package manifests.
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -32,22 +37,34 @@ Swift 5.9
...
+
+  When a package contains a single target, sources may be distributed anywhere within the `./Sources` directory. If sources are placed in a subdirectory under `./Sources/<target>`, or there is more than one target, the existing expectation for sources apply.
+
+* [#6540]
+
+  Build tool plugins can be used with C-family targets
+
+* [#6663]
+
+  Add `visionOS` as a platform alongside `iOS`, `tvOS`, `watchOS`, and the other
```
Comment (MaxDesiatov) [inline CHANGELOG.md:66]:
```suggestion
  Add `visionOS` as a platform alongside `iOS`, `tvOS`, `watchOS`, and the other
  
* [#6023], [#6425], [#6425]

New `swift experimental-sdk` command is now available for managing Swift SDK bundles that follow the format described in [SE-0387]: "Swift SDKs for Cross-Compilation".
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -302,6 +319,8 @@ Swift 3.0
 [SE-0339]: https://github.com/apple/swift-evolution/blob/main/proposals/0339-module-aliasing-for-disambiguation.md
 [SE-0362]: https://github.com/apple/swift-evolution/blob/main/proposals/0362-piecemeal-future-features.md
 [SE-0378]: https://github.com/apple/swift-evolution/blob/main/proposals/0378-package-registry-auth.md
+[SE-0386]: https://github.com/apple/swift-evolution/blob/main/proposals/0386-package-access-modifier.md
```
Comment (MaxDesiatov) [inline CHANGELOG.md:322]:
```suggestion
[SE-0386]: https://github.com/apple/swift-evolution/blob/main/proposals/0386-package-access-modifier.md
[SE-0387]: https://github.com/apple/swift-evolution/blob/main/proposals/0387-cross-compilation-destinations.md
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -32,22 +37,38 @@ Swift 5.9
...
+
+  Build tool plugins can be used with C-family targets
+
+* [#6663]
+
+  Add `visionOS` as a platform alongside `iOS`, `tvOS`, `watchOS`, and the other
+  
+* [#5911], [#5922], [#6023], [#6425]
+
+New `swift experimental-sdk` command is now available for managing Swift SDK bundles that follow the format described in [SE-0387]: "Swift SDKs for Cross-Compilation".
```
Comment (MaxDesiatov) [inline CHANGELOG.md:70]: Formatting fix: lack of this indentation made it render differently.
```suggestion
  New `swift experimental-sdk` command is now available for managing Swift SDK bundles that follow the format described in [SE-0387]: "Swift SDKs for Cross-Compilation".
```

**Comment by MaxDesiatov** [comment]
@swift-ci test windows

File: CHANGELOG.md [hunk 6]
```diff
@@ -3,51 +3,81 @@ Note: This is in reverse chronological order, so newer entries are added to the
...
 
   Remove the `system-module` and `manifest` templates and clean up the remaining `empty`, `library`, and `executable` templates so they include the minimum information needed to get started, with links to documentation in the generated library, executable, and test content.
 
+* [#6185], [#6200]
+
+  Add a new `CompilerPluginSupport` module which contains the definition for macro targets. Macro targets allow authoring and distribution of custom Swift macros such as [expression macros](https://github.com/apple/swift-evolution/blob/main/proposals/0382-expression-macros.md).
+
+* [#6276]
+
+  Add newbuild setting in the package manifest that enables Swift/C++ Interoperability for a given Swift target.
```
Comment (MaxDesiatov) [inline CHANGELOG.md:62]: One more typo nit

```suggestion
  Add new build setting in the package manifest that enables Swift/C++ Interoperability for a given Swift target.
```

**Comment by tomerd** [comment]
@swift-ci smoke test

**Comment by tomerd** [comment]
@swift-ci smoke test windows

**Comment by tomerd** [comment]
@swift-ci smoke test linux

**Comment by MaxDesiatov** [comment]
@swift-ci smoke test lnux

**Comment by MaxDesiatov** [comment]
@swift-ci smoke test linux

**Comment by tomerd** [comment]
@hyp ptal at the description for _interoperablotyMode_ and suggest any changes

File: CHANGELOG.md [hunk 7]
```diff
@@ -3,51 +3,81 @@ Note: This is in reverse chronological order, so newer entries are added to the
...
+  Support for building plugin dependencies for the host when cross-compiling.
+
 * [#6067]
 
   Basic support for a new `.embedInCode` resource rule which allows embedding the contents of the resource into the executable code by generating a byte array, e.g.
 
   ```
   struct PackageResources {
-  static let best_txt: [UInt8] = [104,101,108,108,111,32,119,111,114,108,100,10]
+    static let best_txt: [UInt8] = [104,101,108,108,111,32,119,111,114,108,100,10]
```
Comment (neonichu) [inline CHANGELOG.md:40]: Hm, why is a space here? It definitely won't be there in the generated code :)

**Review by neonichu — APPROVED** [review]

File: CHANGELOG.md (hunk 7 above)
Comment (tomerd) [reply CHANGELOG.md:40]: just to improve formatting?

**Pull request merged**